### PR TITLE
AV-212214 Add URI encoding to rest calls (cherry-pick)

### PR DESCRIPTION
**Pull request by aaha97**

### CODE REVIEW

**Review by ashishnayak96 — APPROVED** [review]
LGTM

**Review by arihantg — APPROVED** [review]

**Comment by akshayhavile** [comment]
build ako

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -622,3 +622,10 @@ func String(s *string) string {
 	}
 	return ""
 }
+
+func GetUriEncoded(uri string) string {
+	if uriSplit := strings.SplitN(uri, "?", 2); len(uriSplit) == 2 {
+		return uriSplit[0] + "?" + url.QueryEscape(uriSplit[1])
```
Comment (sunjayBhatia) [inline pkg/utils/utils.go:628]: this change is not quite right, [url.QueryEscape](https://pkg.go.dev/net/url#QueryEscape) should not be used to escape the whole query string instead it should be used to just escape the value part of each individual query parameter set

in a deployment of ako with this code, we see the following log:

```
lib/avi_api.go:43       msg: Unable to fetch collection data from uri /api/tenant/?name=admin Encountered an error on GET request to URL https://10.83.113.185//api/tenant/?name%3Dadmin: HTTP code: 400; error from Controller: map[error:Invalid Query]
```
the `=` in the `?name=admin` query is getting escaped and should not be

Comment (sunjayBhatia) [reply pkg/utils/utils.go:628]: cc @akshayhavile

**Pull request merged**